### PR TITLE
Wait all sub task return before finish version usage

### PR DESCRIPTION
**Pull request by congqixia**
Related to #26249 
/kind bug

### CODE REVIEW

File: internal/querynodev2/delegator/delegator.go [hunk 1]
```diff
@@ -452,30 +504,36 @@ func executeSubTasks[T any, R interface {
...
-			zap.Error(err),
-		)
-		return nil, err
-	default:
-	}
-
 	results := make([]R, 0, len(tasks))
-	for result := range resultCh {
-		results = append(results, result)
+	for i := 0; i < len(tasks); i++ {
```
Comment (yah01) [inline internal/querynodev2/delegator/delegator.go:514]: this won't work well if the error occurs at the last task

Comment (congqixia) [reply internal/querynodev2/delegator/delegator.go:514]: if err happens at the last task, the following select can only end up in `errCh` branch, which is totally fine

**Review by yah01 — APPROVED** [review]
/lgtm
but /hold first

**Comment by yah01** [comment]
/hold
The problem is, if the first task failed, the others would still perform

**Pull request closed without merging**